Formula/v/vencord-installer.rb

```diff
@@ -0,0 +1,19 @@
+class VencordInstaller < Formula
+  desc "Installer for Vencord"
+  homepage "https://vencord.dev"
+  url "https://github.com/Vencord/Installer/archive/refs/tags/v1.4.0.tar.gz"
+  sha256 "77e476066cf7d249e43416bc0adbf59f38c32eaee207e72419de706d6ba33fc8"
+  license "GPL-3.0-or-later"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-s -w -X 'vencordinstaller/buildinfo.InstallerGitHash=#{tap.user}' -X 'vencordinstaller/buildinfo.InstallerTag=v#{version}'"
+    tags = "static cli"
+    system "go", "build", "-tags", tags, *std_go_args(ldflags:)
+  end
+
+  test do
+    assert_match "Vencord Installer Cli v#{version} (#{tap.user})", shell_output("#{bin}/vencord-installer -version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I included a test case for program failure during execution, as the program relies on Discord.app. However, I am unsure how to incorporate the app into the test.